### DISCUSSION TITLE
Foam template default key, what is the $VARIABLE name?

### DISCUSSION THREAD

**Discussion by vderm** [foambubble/foam#1133] [Q&A]
Hi!

First off, I'm coming over from Spacemacs (emacs) + org-mode + org-roam and what you've done is great! VSCode really feels like the spiritual successor to Emacs.

I'm writing a template and would want to use the name of the wiki link that is used by default and can't figure out what the name is. Tried `FOAM_SELECTED_TEXT`, `FOAM_TITLE` and `FOAM_SLUG` and they either ask for input or it's not it. The [VSCode snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_variables) don't get converted at all.

Basically, when you type `[[document]]` to refer to a file that's not yet created and Ctrl+click (command+click) on the link to create that file, it creates a file called `document.md` and inserts `# document` as the title. I looked through the source code and can't figure out how this is done.

I'm using [Pandoc Citer](https://marketplace.visualstudio.com/items?itemName=notZaki.pandocciter) to insert links to my articles and would want to also have a markdown file per cited article to include the link to the original article, my notes, and so on. This will also allow me (via backlinks), to know which article I've used in which text I've written.

Example:
Citation key is `@OpenAI_LearningDexterousInHand_2019`, so my wikilink would be `[[@OpenAI_LearningDexterousInHand_2019]]` and when I create a note from template, I'd want the created file to be called `@OpenAI_LearningDexterousInHand_2019.md`, be inserted in a subdirectory and the file contains:

``` markdown
---
author:
categories: article
tags:
description:
title: 
---

[[../articles/@OpenAI_LearningDexterousInHand_2019.pdf]]

# @OpenAI_LearningDexterousInHand_2019
```

At some point I'd have to figure out how to extract the author list, tags, title from the bibtex file or from Zotero and put that in the markdown TOML front-matter section, but that's for another day.

Thanks for your help and all the good work!
Vasken

**Comment by riccardoferretti** [comment]
When creating a note from a wikilink, the template used is `new-note.md` and the name of the wikilink becomes the `$FOAM_TITLE` variable.

so, if you have a `.foam/templates/new-note.md` with the following content:

```
# $FOAM_TITLE $FOAM_TITLE
```

and a placeholder link of `[[testing]]`

cmd+clicking on it will create the file `testing.md` with content:

```
# testing testing
```

I hope this is helpful

**Reply by vderm** [reply]
As strange as it is, I made a template in `.foam/templates/new-note.md`:

``` markdown
---
foam_template:
  name: new note
  description: new note
  filepath: 'notes/$FOAM_TITLE.md'
---

---
author:
categories: note
tags:
-
title: $FOAM_TITLE
---

# $FOAM_TITLE
```

And while in a markdown file, I can cmd+click on a placeholder `[[@OpenAI_LearningDexterousInHand_2019]]` and it creates it properly without prompting for an input:

``` markdown
---
author:
categories: note
tags:
-
title: @OpenAI_LearningDexterousInHand_2019
---

# @OpenAI_LearningDexterousInHand_2019
```

But when I create a note from a template (either with the new-note template or another), it does ask for me to input a title for the file.

Anyway, I'll just use the new-note template and the `${1:|note,article|}` option selection feature to pick what the note is for.

Thanks for your help!

**Reply by riccardoferretti** [reply]
Basically `$FOAM_TITLE` is a variable that is requested to the user if not already provided to the template.
- When you create a new note by clicking on a placeholder, we populate `$FOAM_TITLE` with the content of the placeholder.
- When you create a new note from the command palette, `$FOAM_TITLE` has no value, so it will be asked (if present in the template being used).

More info here: https://github.com/foambubble/foam/blob/master/docs/user/features/note-templates.md#variables
(although I admit the documentation is not super clear in this regard)
